java/src/main/java/com/genexus/db/driver/DataSourceConnectionPool.java

```diff
@@ -119,6 +119,31 @@ public void disconnect() throws SQLException
 			en.nextElement().disconnect();
 	}
 	@Override
+	public void flushBuffers(int handle, java.lang.Object o) throws SQLException
+	{
+		for (Enumeration<ConnectionPool> en = readWritePools.elements(); en.hasMoreElements(); )
+		{
+			ConnectionPool connPool = en.nextElement();
+			for (Enumeration en1 = connPool.getConnections(); en1.hasMoreElements(); ) {
+				GXConnection conn = (GXConnection) en1.nextElement();
+				if (conn.getHandle() == handle) {
+					conn.flushBatchCursors(o);
+				}
+			}
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 124:133, 135:144.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.